tests/test_clients_GcpClient.py

```diff
@@ -0,0 +1,249 @@
+                return HttpRequest(
+                    http,
+                    model.response,
+                    uri,
+                    method=method,
+                    headers={}
+                )
+            elif snapshot == not_found_snapshot_name:
+                http = HttpMock('tests/data/snapshots.get.notfound.json', {'status': '404'})
+                model = JsonModel()
+                uri = 'some_uri'
+                method = 'GET'
+                return HttpRequest(
+                    http,
+                    model.response,
+                    uri,
+                    method=method,
+                    headers={}
+                )
+            elif snapshot == invalid_snapshot_name:
+                http = HttpMock('tests/data/snapshots.get.forbidden.json', {'status': '403'})
+                model = JsonModel()
+                uri = 'some_uri'
+                method = 'GET'
+                return HttpRequest(
+                    http,
+                    model.response,
+                    uri,
+                    method=method,
+                    headers={}
+                )
+
+    class disks:
+        def get(self, project, zone, disk):
+            if disk == valid_disk_name:
+                http = HttpMock('tests/data/disks.get.json', {'status': '200'})
+                model = JsonModel()
+                uri = 'some_uri'
+                method = 'GET'
+                return HttpRequest(
+                    http,
+                    model.response,
+                    uri,
+                    method=method,
+                    headers={}
+                )
+            elif disk == not_found_disk_name:
+                http = HttpMock('tests/data/disks.get.notfound.json', {'status': '404'})
+                model = JsonModel()
+                uri = 'some_uri'
+                method = 'GET'
+                return HttpRequest(
+                    http,
+                    model.response,
+                    uri,
+                    method=method,
+                    headers={}
+                )
+            elif disk == invalid_disk_name:
+                http = HttpMock('tests/data/disks.get.forbidden.json', {'status': '403'})
+                model = JsonModel()
+                uri = 'some_uri'
+                method = 'GET'
+                return HttpRequest(
+                    http,
+                    model.response,
+                    uri,
+                    method=method,
+                    headers={}
+                )
+
+class StorageClient:
+    def get_bucket(self, container):
+        if container == valid_container:
+            return bucket
+        elif container == invalid_container:
+            raise NotFound()
+
+class Blob:
+    def upload_from_string(self, data, content_type='text/plain', client=None, predefined_acl=None):
+        pass
+    
+    def delete(self, client=None):
+        return self
+
+class TestGcpClient:
```

Comment:
Is it possible to break down the monolithic class for GCP Tests. We might need to add a lot of tests going forward. So, it can be considered controller class and for several types of tests we can write classes? Just thinking out loud!